### ISSUE TITLE
Make FHIR proxy server context path customizable

### ISSUE THREAD

**Issue by holger-stenzhorn** [datasharingframework/dsf#85]
The context path of the FHIR proxy server is currently hardcoded to `/fhir`. To render this customizable, only a few changes need to be performed:

- Add an environment variable `APP_SERVER_CONTEXT_PATH` and default it to `/fhir` in `dsf-docker/fhir_proxy/Dockerfile`:
  ```diff
  ENV PROXY_PASS_CONNECTION_TIMEOUT_WS=30
  +ENV APP_SERVER_CONTEXT_PATH="/fhir"
  ```
- Replace the hardcoded context path `/fhir` in `dsf-docker/fhir_proxy/conf/extra/host-ssl.conf` with that environment variable:
  ```diff
  -<Location "/fhir">
  +<Location "${APP_SERVER_CONTEXT_PATH}">
          RequestHeader set X-ClientCert %{SSL_CLIENT_CERT}s
 
  -       ProxyPass http://${APP_SERVER_IP}:8080/fhir timeout=${PROXY_PASS_TIMEOUT_HTTP} connectiontimeout=${PROXY_PASS_CONNECTION_TIMEOUT_HTTP}
  -       ProxyPassReverse http://${APP_SERVER_IP}:8080/fhir
  +       ProxyPass http://${APP_SERVER_IP}:8080${APP_SERVER_CONTEXT_PATH} timeout=${PROXY_PASS_TIMEOUT_HTTP} connectiontimeout=${PROXY_PASS_CONNECTION_TIMEOUT_HTTP}
  +       ProxyPassReverse http://${APP_SERVER_IP}:8080${APP_SERVER_CONTEXT_PATH}
   </Location>
  -<Location "/fhir/ws">
  +<Location "${APP_SERVER_CONTEXT_PATH}/ws">
          RequestHeader set X-ClientCert %{SSL_CLIENT_CERT}s
 
          ProxyWebsocketFallbackToProxyHttp off
  -       ProxyPass ws://${APP_SERVER_IP}:8080/fhir/ws timeout=${PROXY_PASS_TIMEOUT_WS} connectiontimeout=${PROXY_PASS_CONNECTION_TIMEOUT_WS}
  -       ProxyPassReverse ws://${APP_SERVER_IP}:8080/fhir/ws
  +       ProxyPass ws://${APP_SERVER_IP}:8080${APP_SERVER_CONTEXT_PATH}/ws timeout=${PROXY_PASS_TIMEOUT_WS} connectiontimeout=${PROXY_PASS_CONNECTION_TIMEOUT_WS}
  +       ProxyPassReverse ws://${APP_SERVER_IP}:8080${APP_SERVER_CONTEXT_PATH}/ws
   </Location>
  ```
- Add the following (exemplarily) then to `docker-compose.yml`:
  (Note that the environment variable `DEV_DSF_SERVER_CONTEXT_PATH` already exists in `dsf-fhir/dsf-fhir-server-jetty/docker/Dockerfile`)
  ```diff
  services:
    proxy:
      image: ghcr.io/datasharingframework/fhir_proxy:1.1.0
      ...
      environment:
        ...
  +     APP_SERVER_CONTEXT_PATH: /dsf/fhir
    ...
    app:
      image: ghcr.io/datasharingframework/fhir:1.1.0
      ...
      environment:
        ...
  +     DEV_DSF_SERVER_CONTEXT_PATH: /dsf/fhir
    ...
  ```

The above has been implemented and successfully tested in our local environment already.

⚠️ There just seems to be a tiny bug though in `dsf-fhir/dsf-fhir-server/src/main/java/dev/dsf/fhir/adapter/HtmlFhirAdapter.java` in `private String getUrlHeading(Resource resource)` which leads to a a double display of `/fhir` in the shown title, e.g. `https://diz.uks.eu/dsf/fhir/fhir/metadata` and  - if I am not mistaken - there should be a `html` tab left to the `json` and `xml` tabs. (see screenshot)

![temp](https://github.com/datasharingframework/dsf/assets/8993693/2ea20ea8-e445-43c5-bc29-20e220614b4c)

**Comment by hhund** [comment]
@holger-stenzhorn I have implemented your suggestion slightly modified in #95. The `DEV_DSF_SERVER_CONTEXT_PATH` variable does not need to be specified for the DSF FHIR app server if the apache httpd `ProxyPass` and `ProxyPassReverse` directives stay hard coded to the DSF FHIR app server context path default ...`/fhir`...

**Comment by holger-stenzhorn** [comment]
The given solution works nicely but there is a tiny "edge case": If you enter e.g. `https://diz.uks.eu/dsf/test/fhir` then this gets redirected to `https://diz.uks.eu/fhir`. A quick and easy fix for this issue would be to change `host-ssl.conf` accordingly:
```diff
+Redirect ${SERVER_CONTEXT_PATH} ${SERVER_CONTEXT_PATH}/
-<Location "${SERVER_CONTEXT_PATH}">
+<Location "${SERVER_CONTEXT_PATH}/">
        RequestHeader set X-ClientCert %{SSL_CLIENT_CERT}s

-       ProxyPass http://${APP_SERVER_IP}:8080/fhir timeout=${PROXY_PASS_TIMEOUT_HTTP} connectiontimeout=${PROXY_PASS_CONNECTION_TIMEOUT_HTTP}
-       ProxyPassReverse http://${APP_SERVER_IP}:8080/fhir
+       ProxyPass http://${APP_SERVER_IP}:8080/fhir/ timeout=${PROXY_PASS_TIMEOUT_HTTP} connectiontimeout=${PROXY_PASS_CONNECTION_TIMEOUT_HTTP}
+       ProxyPassReverse http://${APP_SERVER_IP}:8080/fhir/
```
